server/spawn-vehicle.lua

```diff
@@ -13,46 +13,67 @@ end
     local accessPoint = garage.accessPoints[accessPointIndex]
-    if #(GetEntityCoords(GetPlayerPed(source)) - accessPoint.coords.xyz) > 3 then
-        lib.print.error(string.format("player %s attempted to spawn a vehicle but was too far from the access point", source))
-        return
-    end
     local garageType = GetGarageType(garageName)
-
     local spawnCoords = accessPoint.spawn or accessPoint.coords
-    if Config.distanceCheck then
-        local vec3Coords = vec3(spawnCoords.x, spawnCoords.y, spawnCoords.z)
-        local nearbyVehicle = lib.getClosestVehicle(vec3Coords, Config.distanceCheck, false)
-        if nearbyVehicle then
-            exports.qbx_core:Notify(source, locale('error.no_space'), 'error')
-            return
-        end
-    end
+
+    local hookPayload = {
+        source = source,
+        vehicleId = vehicleId,
+        garage = garage,
+        garageName = garageName,
+        accessPointIndex = accessPointIndex,
+        garageType = garageType,
+        spawnCoords = spawnCoords,
+        distanceCheck = Config.distanceCheck,
+        doorsLocked = Config.doorsLocked,
+        warpPed = Config.warpInVehicle,
+        vehiclekeys = Config.giveKeys,
+    }
 
     local filter = GetPlayerVehicleFilter(source, garageName)
     local playerVehicle = exports.qbx_vehicles:GetPlayerVehicle(vehicleId, filter)
     if not playerVehicle then
         exports.qbx_core:Notify(source, locale('error.not_owned'), 'error')
         return
     end
-    if garageType == GarageType.DEPOT and FindPlateOnServer(playerVehicle.props.plate) then -- If depot, check if vehicle is not already spawned on the map
-        return exports.qbx_core:Notify(source, locale('error.not_impound'), 'error', 5000)
+
+    hookPayload.playerVehicle = playerVehicle
+
+    if GaragesHooks('spawnVehicle', hookPayload) == false then
+        lib.print.debug("Vehicle spawn was canceled by a hook.")
+        return
     end
 
-    local warpPed = Config.warpInVehicle and GetPlayerPed(source)
-    local netId, veh = qbx.spawnVehicle({ spawnSource = spawnCoords, model = playerVehicle.props.model, props = playerVehicle.props, warp = warpPed})
+    local warpPed = hookPayload.warpPed and GetPlayerPed(source)
+    local netId, veh = qbx.spawnVehicle({
+        spawnSource = spawnCoords,
+        model = playerVehicle.props.model,
+        props = playerVehicle.props,
+        warp = warpPed
+    })
+
+    hookPayload.netId = netId
+    hookPayload.veh = veh
 
-    if Config.doorsLocked then
+    if GaragesHooks('spawnedVehicle', hookPayload) == false then
+        DeleteEntity(veh)
+        return
+    end
+
+    if hookPayload.doorsLocked then
         SetVehicleDoorsLocked(veh, 2)
     end
 
-    TriggerClientEvent('vehiclekeys:client:SetOwner', source, playerVehicle.props.plate)
+    if hookPayload.vehiclekeys then
+        TriggerClientEvent('vehiclekeys:client:SetOwner', source, playerVehicle.props.plate)
+    end
```

Comment:
Do not use a hook to control garages in this way. The hooks should only be used to cancel the event.

Reply:
vehiclekeys and link to the config